src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const obj = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (true) {
```

Comment:
use switch (action.type) {
case 'addProperties': ...
instead of switch (true) {
case action.type === 'addProperties':...
because you repeat the code 'action.type === '  3 times